src/assemble_files.py

```diff
@@ -32,7 +32,7 @@ def create_outname(outdir, infile):
 def assemble(infile, outfile, extras):
   """Create the command-line for an assembler invocation."""
   assembler = extras['assembler']
-  basename = os.path.basename(assembler)
+  basename = os.path.splitext(os.path.basename(assembler))[0]
   commands = {
       'wast2wasm': [extras['assembler'], infile, '-o', outfile],
```

Comment:
Existing nit: should these two `extras['assembler']` references be just `assembler`?